File: data/derived_cot/rq2_eval/translation/r1/_converted_code/java_to_py/AreaCalculator.py
import math

class AreaCalculator:

    def __init__(self, radius):
        self.radius = radius

    def calculate_circle_area(self):
        return math.pi * math.pow(self.radius, 2)

    def calculate_sphere_area(self):
        return 4 * math.pi * math.pow(self.radius, 2)

    def calculate_cylinder_area(self, height):
        return 2 * math.pi * self.radius * (self.radius + height)

    def calculate_sector_area(self, angle):
        return math.pow(self.radius, 2) * angle / 2

    def calculate_annulus_area(self, innerRadius, outerRadius):
        return math.pi * (math.pow(outerRadius, 2) - math.pow(innerRadius, 2))

def main():
    area_calculator = AreaCalculator(2.0)
    print('Circle Area:', area_calculator.calculateCircleArea())
    print('Sphere Area:', area_calculator.calculateSphereArea())
    print('Cylinder Area:', area_calculator.calculateCylinderArea(3.0))
    print('Sector Area:', area_calculator.calculateSectorArea(math.pi))
    print('Annulus Area:', area_calculator.calculateAnnulusArea(2.0, 3.0))
if __name__ == '__main__':
    main()

File: data/derived_cot/rq3_prompting/results/translation/qwen/java_to_cpp/code_output/AreaCalculator.cpp
#include <cmath>
#include <iostream>

class AreaCalculator {
private:
    double radius;
    static constexpr double PI { std::acos(0.0) * 2.0 };

public:
    explicit AreaCalculator(double radius) : radius(radius) {}

    double calculateCircleArea() const {
        return PI * std::pow(radius, 2);
    }

    double calculateSphereArea() const {
        return 4 * PI * std::pow(radius, 2);
    }

    double calculateCylinderArea(double height) const {
        return 2 * PI * radius * (radius + height);
    }

    double calculateSectorArea(double angle) const {
        return std::pow(radius, 2) * angle / 2;
    }

    double calculateAnnulusArea(double innerRadius, double outerRadius) const {
        return PI * (std::pow(outerRadius, 2) - std::pow(innerRadius, 2));
    }
};

int main() {
    AreaCalculator areaCalculator(2);
    std::cout << "Circle Area: " << areaCalculator.calculateCircleArea() << '\n';
    std::cout << "Sphere Area: " << areaCalculator.calculateSphereArea() << '\n';
    std::cout << "Cylinder Area: " << areaCalculator.calculateCylinderArea(3) << '\n';
    std::cout << "Sector Area: " << areaCalculator.calculateSectorArea(AreaCalculator::PI) << '\n';
    std::cout << "Annulus Area: " << areaCalculator.calculateAnnulusArea(2, 3) << '\n';
    return 0;
}